lib/glossarist/glossary_definition.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require "yaml"
+
+module Glossarist
+  class GlossaryDefinition
+    config = YAML.load_file(File.expand_path("../../../config.yml", __FILE__)) || {}
+
+    DEFAULT_CONFIG = {
+      "concept_source" => {
+        "status" => %w[
+          identical
+          modified
+          restyle
+          context-added
+          generalisation
+          specialisation
+          unspecified
+        ],
+
+        "type" => %w[
+          authoritative
+          lineage
+        ],
+      },
+
+      "related_concept" => {
+        "type" => %w[
+          deprecates
+          supersedes
+          superseded_by
+          narrower
+          broader
+          equivalent
+          compare
+          contrast
+          see
+        ],
+      },
+
+      "abbreviation" => {
+        "type" => %w[
+          truncation
+          acronym
+          initialism
+        ],
+      },
+
+      "grammar_info" => {
+        "boolean_attribute" => %w[
+          preposition
+          participle
+          adj
+          verb
+          adverb
+          noun
+        ],
+
+        # m => masculine
+        # f => feminine
+        # n => neuter
+        # c => common
+        # using initial letter because this is how it is used in iev-data
+        "gender" => %w[
+          m
+          f
+          n
+          c
+        ],
+
+        "number" => %w[
+          singular
+          dual
+          plural
+        ],
+      },
+
+      "designation" => {
+        "normative_status" => %w[
+          preferred
+          admitted
+          deprecated
+        ],
+      },
+    }.freeze
+
+    CONCEPT_SOURCE_STATUSES = (
+      config.dig("concept_source", "status") ||
+      DEFAULT_CONFIG.dig("concept_source", "status")
```

Comment:
I am thinking to refactor this approach and totally rely on `config.yml` and move the default config to `config.yml`. For that change, we will have to update the config file for tests related to config and then after the tests, we will have to revert the file to its original state.

@ronaldtse what do you think about this?

Reply:
@HassanAkbar I think that could work. We can try?